src/lib.rs

```diff
@@ -198,119 +198,148 @@ mod sym;
+    ///     ...
+    /// }
+    /// # };
+    /// ```
+    ///
+    /// # Canonical entry point:
+    ///
+    /// ```
+    /// # const IGNORE: &str = stringify! {
+    /// extern crate proc_macro;
+    ///
+    /// use proc_macro::TokenStream;
+    ///
+    /// static WASM: watt::Instance = watt::Instance::new(include_bytes!("my_macro.wasm"));
+    ///
+    /// #[proc_macro_derive(MyDerive)]
+    /// pub fn my_macro(input: TokenStream) -> TokenStream {
+    ///     WASM.proc_macro_derive("my_macro", input)
+    /// }
+    /// # };
+    /// ```
+    pub fn proc_macro_derive(&self, fun: &str, input: TokenStream) -> TokenStream {
+        exec::proc_macro(fun, vec![input], self)
+    }
+
+    /// A #\[proc_macro_attribute\] implemented in wasm!
+    ///
+    /// # Canonical macro implementation:
+    ///
+    /// ```
+    /// # const IGNORE: &str = stringify! {
+    /// use proc_macro2::TokenStream;
+    ///
+    /// #[proc_macro2::proc_macro_attribute]
+    /// pub fn my_macro(args: TokenStream, input: TokenStream) -> TokenStream {
+    ///     ...
+    /// }
+    /// # };
+    /// ```
+    ///
+    /// # Canonical entry point:
+    ///
+    /// ```
+    /// # const IGNORE: &str = stringify! {
+    /// extern crate proc_macro;
+    ///
+    /// use proc_macro::TokenStream;
+    ///
+    /// static WASM: watt::Instance = watt::Instance::new(include_bytes!("my_macro.wasm"));
+    ///
+    /// #[proc_macro_attribute]
+    /// pub fn my_macro(args: TokenStream, input: TokenStream) -> TokenStream {
+    ///     WASM.proc_macro_attribute("my_macro", args, input)
+    /// }
+    /// # };
+    /// ```
+    pub fn proc_macro_attribute(
+        &self,
+        fun: &str,
+        args: TokenStream,
+        input: TokenStream,
+    ) -> TokenStream {
+        exec::proc_macro(fun, vec![args, input], self)
+    }
 
-/// A #\[proc_macro_attribute\] implemented in wasm!
-///
-/// # Canonical macro implementation:
-///
-/// ```
-/// # const IGNORE: &str = stringify! {
-/// use proc_macro2::TokenStream;
-///
-/// #[no_mangle]
-/// pub extern "C" fn my_macro(args: TokenStream, input: TokenStream) -> TokenStream {
-///     proc_macro2::set_wasm_panic_hook();
-///
-///     ...
-/// }
-/// # };
-/// ```
-///
-/// # Canonical entry point:
-///
-/// ```
-/// # const IGNORE: &str = stringify! {
-/// extern crate proc_macro;
-///
-/// use proc_macro::TokenStream;
-///
-/// static WASM: &[u8] = include_bytes!("my_macro.wasm");
-///
-/// #[proc_macro_attribute]
-/// pub fn my_macro(args: TokenStream, input: TokenStream) -> TokenStream {
-///     watt::proc_macro_attribute("my_macro", args, input, WASM)
-/// }
-/// # };
-/// ```
-pub fn proc_macro_attribute(
-    fun: &str,
-    args: TokenStream,
-    input: TokenStream,
-    wasm: &[u8],
-) -> TokenStream {
-    exec::proc_macro(fun, vec![args, input], wasm)
+    fn id(&self) -> usize {
+        static NEXT_ID: AtomicUsize = AtomicUsize::new(1);
+        match self.id.load(SeqCst) {
```

Comment:
I'm fairly confident the orderings in this method can be `Relaxed`, as they aren't providing any ordering guarantees for memory other than themselves.

Reply:
I personally prefer to use `SeqCst` everywhere unless performance becomes an issue, and I'm pretty certain performance is not an issue here.